contracts/registries/PoolRegistry.sol

```diff
@@ -0,0 +1,350 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "../interfaces/ISwap.sol";
+import "../interfaces/IMetaSwap.sol";
+import "../meta/MetaSwapDeposit.sol";
+
+/**
+ * @title PoolRegistry
+ * @notice This contract holds list of pools deployed.
+ */
+contract PoolRegistry is Ownable, AccessControl {
+    using SafeMath for uint256;
+
+    bytes32 public constant SADDLE_MANAGER_ROLE =
+        keccak256("SADDLE_MANAGER_ROLE");
+    bytes32 public constant COMMUNITY_MANAGER_ROLE =
+        keccak256("COMMUNITY_MANAGER_ROLE");
+
+    PoolData[] public saddlePools;
+    mapping(address => uint256) public poolToSaddleIndexPlusOne;
```

Comment:
Since we only care about these for viewing, there's no difference in cost between storing a saddle-only pool, and having a view function which iterates all pools and only returns the ones where `isSaddle=true`, right?